src/Sulu/Bundle/RouteBundle/Controller/RouteController.php

```diff
@@ -191,4 +208,19 @@ private function generateUrlResponse(Request $request)
 
         return $this->handleView($this->view(['resourcelocator' => $route]));
     }
+
+    private function generateUrl(string $locale, array $parts, array $options): string
+    {
+        if (!$this->translator || !$this->translator instanceof LocaleAwareInterface){
+            return $this->routeGenerator->generate($parts, $options);
```

Comment:
what do you think about passing the `locale` as option inside of the `$options` parameter here? that way we dont need to think about this in other places that use the `RouteGenerator`

Reply:
think it would be nice to move the handling of the locale from the entity to `RouteGenerator` too when doing this:
https://github.com/sulu/sulu/blob/5c7a15c250c01ee4337f1fd1c44a34b2e9035147/src/Sulu/Bundle/RouteBundle/Generator/SymfonyExpressionTokenProvider.php#L57-L59

Reply:
I updated the PR to adding the locale to the options which us give to the generator.

Reply:
🎉